sql/core/src/main/scala/org/apache/spark/sql/execution/joins/HashedRelation.scala

```diff
@@ -311,6 +314,15 @@ private[joins] object UnsafeHashedRelation {
       key: Seq[Expression],
       sizeEstimate: Int,
       taskMemoryManager: TaskMemoryManager): HashedRelation = {
+    apply(input, key, sizeEstimate, taskMemoryManager, isNullAware = false)
+  }
+
+  def apply(
+      input: Iterator[InternalRow],
+      key: Seq[Expression],
+      sizeEstimate: Int,
+      taskMemoryManager: TaskMemoryManager,
+      isNullAware: Boolean = false): HashedRelation = {
```

Comment:
do we really need 2 `apply` methods as we have default parameter value here?

Reply:
let me remove them and see if there are build issues. if not , I will remove them for good.